src/cart/crt_hg.c

```diff
@@ -904,10 +904,16 @@ crt_hg_class_init(crt_provider_t provider, int ctx_idx, bool primary, int iface_
 	if (prov_data->cpg_max_unexp_size > 0)
 		init_info.na_init_info.max_unexpected_size = prov_data->cpg_max_unexp_size;
 
-	init_info.request_post_init = crt_gdata.cg_post_init;
-	init_info.request_post_incr = crt_gdata.cg_post_incr;
-
-	hg_class = HG_Init_opt(info_string, crt_is_service(), &init_info);
+	init_info.request_post_init         = crt_gdata.cg_post_init;
+	init_info.request_post_incr         = crt_gdata.cg_post_incr;
+	init_info.multi_recv_op_max         = crt_gdata.cg_mrecv_buf;
+	init_info.multi_recv_copy_threshold = crt_gdata.cg_mrecv_buf_copy;
+	/* With cxi provider, separate SWIM traffic in an effort to prevent potential congestion. */
+	if (provider == CRT_PROV_OFI_CXI && crt_is_service() &&
+	    crt_gdata.cg_auto_swim_disable == 0 && ctx_idx == crt_gdata.cg_swim_ctx_idx)
```

Comment:
the check for cg_auto_swim_disable should not be done; it only tells you whether swim is to be enabled at startup, as opposed to being enabled manually after the context has been created (like some tests do). In both cases you would want to use a separate class.

Reply:
OK I'll fix it after I rebase.